File: src/main/javascript/examples/device_bind.js
import fetch from 'node-fetch';
import crypto from 'crypto';
import {base_url, deviceNo, userId, key} from './constant.js'

const timestamp = Date.now().toString();
const token = crypto.createHash('md5').update(key + timestamp).digest('hex');

fetch(base_url + '/en/device/' + deviceNo + '/open', {
    method: 'POST',
    headers: {
        'time': timestamp,
        'token': token,
        'userId': userId,
        'Content-Type': 'application/json'
    }
})
    .then(response => response.json())
    .then(data => console.log(data))
    .catch(error => console.error('Error:', error));

File: src/main/python/examples/device_bind.py
import hashlib
import time
import requests
from constant import base_url, device_no, user_id, key

timestamp = str(int(time.time() * 1000))
token = hashlib.md5((key + timestamp).encode()).hexdigest()

url = f"{base_url}/en/device/{device_no}/open"
headers = {
    "time": timestamp,
    "token": token,
    "userId": user_id,
    "Content-Type": "application/json"
}

response = requests.post(url, headers=headers)
print(response.json())
